ios/RNSnowplowTracker.m

```diff
@@ -14,175 +14,216 @@ - (dispatch_queue_t)methodQueue
+            NSError * error = [NSError errorWithDomain:@"SnowplowTracker" code:100 userInfo:nil];
+            reject(@"ERROR", @"SnowplowTracker: initiasetSubjectDatalize() method -  colorDepth cannot be null", error);
+        } else {
+            [self.tracker.subject setColorDepth:[options[@"colorDepth"] integerValue]];
+        }
+    }
 }
 
 RCT_EXPORT_METHOD(trackSelfDescribingEvent
                   :(nonnull SPSelfDescribingJson *)event
                   :(NSArray<SPSelfDescribingJson *> *)contexts) {
+
     SPUnstructured * unstructEvent = [SPUnstructured build:^(id<SPUnstructuredBuilder> builder) {
         [builder setEventData:event];
         if (contexts) {
             [builder setContexts:[[NSMutableArray alloc] initWithArray:contexts]];
         }
     }];
+
     [self.tracker trackUnstructuredEvent:unstructEvent];
 }
 
 RCT_EXPORT_METHOD(trackStructuredEvent
-                  :(nonnull NSString *)category // required (non-empty string)
-                  :(nonnull NSString *)action // required
-                  :(NSString *)label
-                  :(NSString *)property
-                  :(double)value
+                  :(NSDictionary *)details
                   :(NSArray<SPSelfDescribingJson *> *)contexts) {
+
     SPStructured * trackerEvent = [SPStructured build:^(id<SPStructuredBuilder> builder) {
-        [builder setCategory:category];
-        [builder setAction:action];
-        [builder setValue:value];
-        if (label != nil) [builder setLabel:label];
-        if (property != nil) [builder setProperty:property];
+        [builder setCategory:details[@"category"]];
+        [builder setAction:details[@"action"]];
+        if (details[@"label"] != nil) {
+            [builder setLabel:details[@"label"]];
+        }
+        if (details[@"property"] != nil) {
+            [builder setProperty:details[@"property"]];
+        }
+        // doubleValue cannot be NSNull, and falsey value evaluates to 0 in objective-c. Only set 'value' parameter where neither are the case.
+        if (details[@"value"] != (id)[NSNull null] && details[@"value"] != nil) {
+            [builder setValue:[details[@"value"] doubleValue]];
+        }
         if (contexts) {
             [builder setContexts:[[NSMutableArray alloc] initWithArray:contexts]];
         }
     }];
+
     [self.tracker trackStructuredEvent:trackerEvent];
 }
 
 RCT_EXPORT_METHOD(trackScreenViewEvent
-                  :(nonnull NSString *)screenName
-                  :(NSString *)screenId
-                  :(NSString *)screenType
-                  :(NSString *)previousScreenName
-                  :(NSString *)previousScreenType
-                  :(NSString *)previousScreenId
-                  :(NSString *)transitionType
+                  :(NSDictionary *)details
                   :(NSArray<SPSelfDescribingJson *> *)contexts) {
+
     SPScreenView * SVevent = [SPScreenView build:^(id<SPScreenViewBuilder> builder) {
```

Comment:
variable better lowercased.
On ObjC the convention is long self-explaining names so it would be perfectly fine `screenViewEvent`